src/Fixture/Factory/SettingsFixtureFactory.php

```diff
@@ -0,0 +1,120 @@
+        $this->settingsFormatter = $settingsFormatter;
+        $this->optionsResolver = new OptionsResolver();
+
+        $this->configureOptions($this->optionsResolver);
+    }
+
+    public function create(array $options = []): SettingInterface
+    {
+        $options = $this->optionsResolver->resolve($options);
+
+        /** @var SettingsInterface $settings */
+        $settings = $this->settingsRegistry->getByAlias($options['alias']);
+        ['vendor' => $vendor, 'plugin' => $plugin] = $settings->getAliasAsArray();
+
+        /** @var SettingInterface|null $setting */
+        $setting = $this->settingRepository->findOneBy([
+            'vendor' => $vendor,
+            'plugin' => $plugin,
+            'path' => $options['path'],
+            'localeCode' => $options['locale'],
+            'channel' => $options['channel'],
+        ]);
+
+        if (null === $setting) {
+            /** @var SettingInterface $setting */
+            $setting = $this->settingFactory->createNew();
+            $setting->setVendor($vendor);
+            $setting->setPlugin($plugin);
+            $setting->setPath($options['path']);
+            $setting->setLocaleCode($options['locale']);
+            if (null !== $options['channel']) {
+                /** @var ?ChannelInterface $channel */
+                $channel = $this->channelRepository->findOneBy(['code' => $options['channel']]);
+                $setting->setChannel($channel);
+            }
+            $setting->setStorageType($options['type']); // Do it once for the reset of the previous value to work
+        } elseif ($options['ignore_if_exists']) {
+            return $setting;
+        }
+
+        $setting->setValue(null); // reset the previous value according to the potential previous type
+        $setting->setStorageType($options['type']); // If the type has changed, we change it!
+        $setting->setValue($this->settingsFormatter->formatValue($options['type'], $options['value']));
+
+        return $setting;
+    }
+
+    protected function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setDefault('alias', '')
+            ->setAllowedTypes('alias', 'string')
+            ->setDefault('path', '')
+            ->setAllowedTypes('path', 'string')
+            ->setDefault('channel', null)
+            ->setAllowedTypes('channel', ['null', 'string'])
+            ->setDefault('locale', null)
+            ->setAllowedTypes('locale', ['null', 'string'])
+            ->setDefault('type', SettingInterface::STORAGE_TYPE_TEXT)
+            ->setAllowedTypes('type', 'string')
+            ->setAllowedValues('type', [SettingInterface::STORAGE_TYPE_TEXT, SettingInterface::STORAGE_TYPE_BOOLEAN, SettingInterface::STORAGE_TYPE_INTEGER,
+                SettingInterface::STORAGE_TYPE_FLOAT, SettingInterface::STORAGE_TYPE_JSON, SettingInterface::STORAGE_TYPE_DATE, SettingInterface::STORAGE_TYPE_DATETIME])
```

Comment:
**🎯 suggestion:** We could have a static method into the SettingInterface implementation returning all the types? Like `Setting::getAllStorageTypes():array`.‏